graylog2-server/src/main/java/org/graylog2/shared/email/EmailFactory.java

```diff
@@ -119,6 +120,9 @@ private <T extends Email> T applyConfig(T email) throws EmailException {
             email.setFrom(configuration.getFromEmail());
         }
 
+        email.setSocketConnectionTimeout(Ints.saturatedCast(configuration.getSocketConnectionTimeout().toMillis()));
+        email.setSocketTimeout(Ints.saturatedCast(configuration.getSocketTimeout().toMillis()));
+
```

Comment:
can this be a problem for the user that the validation is always done using longs, however this is set to a integer? 
I know that durations larger than Integer.MAX_VALUE won't be seen in reality, so I'm fine with that in any way.

Reply:
I only added the validation to fail fast and not wait until the first email is sent, but I thought it was ok to silently limit the timeout here to ~24 hours. It doesn't make much sense anyway to configure such a high value. It looks like we are doing this at a couple of places already when dealing with durations.